tests/models/test_pca.py

```diff
@@ -9,20 +11,21 @@
 N_small = 128
 seed = 83
 small_rank = 16
-set_all_random_seeds(seed)
```

Comment:
I think it makes sense to keep it here. What's the reason for moving?

Reply:
Actually, I take it back. This can potentially mess up other tests.  Is there an easy way to "unset" the seed? That way we can set them on startup and unset them on teardown.

Reply:
That was actually my reason for moving it under the functions that create the data matrix, because I thought doing this would make sure the seed is only set locally. But I just checked and you are right, calling it once make all the random functions that follow deterministic. I'm looking into possible ways to only set the seed temporarily

Reply:
You can create an `unset_all_random_seeds` and set them to `None`, which will revert the change. Then you can make a pytest set up function that sets the seeds and a teardown function that unsets them (see [here](https://medium.com/@ramanish1992/pytest-setup-tear-down-configuration-c8197558a8e2) for an example)

Reply:
Can you take a look at the latest changes? I think this should work now...